rclcpp/test/rclcpp/node_interfaces/test_node_type_descriptions.cpp

```diff
@@ -37,6 +37,18 @@ TEST_F(TestNodeTypeDescriptions, interface_created)
   ASSERT_NE(nullptr, node.get_node_type_descriptions_interface());
 }
 
+TEST_F(TestNodeTypeDescriptions, automatically_declare_parameters_from_overrides)
+{
+  rclcpp::NodeOptions node_options;
+  node_options.automatically_declare_parameters_from_overrides(true);
+  node_options.append_parameter_override("start_type_description_service", false);
+  ASSERT_NO_THROW(
+  {
+    auto node = std::make_shared<rclcpp::Node>("node", "ns", node_options);
+    (void) node;
+  });
```

Comment:
implementation looks good to me, i think this is a right thing to do instead of silently allowing the user application to declare `start_type_description_service` in any parameter types.

probably we would want to add the `EXPECT_THROW` with `std::invalid_argument` test case? that is the original problem raised by https://github.com/ros2/rclcpp/issues/2892?